spec/forms/support_interface/application_forms/reinstate_declined_offer_form_spec.rb

```diff
@@ -0,0 +1,57 @@
+require 'rails_helper'
+
+RSpec.describe SupportInterface::ApplicationForms::ReinstateDeclinedOfferForm, type: :model, with_audited: true do
+  describe 'validations' do
+    it { is_expected.to validate_presence_of(:accept_guidance) }
+    it { is_expected.to validate_presence_of(:audit_comment_ticket) }
+
+    context 'for an invalid zendesk link' do
+      invalid_link = 'nonsense'
+      it { is_expected.not_to allow_value(invalid_link).for(:audit_comment_ticket) }
+    end
+
+    context 'for an valid zendesk link' do
+      valid_link = 'www.becomingateacher.zendesk.com/agent/tickets/example'
+      it { is_expected.to allow_value(valid_link).for(:audit_comment_ticket) }
+    end
+  end
+
+  describe '.build_from_course_choice' do
+    it 'creates an object based on the provided ApplicationChoice' do
+    end
+  end
+
+  describe '#save' do
+    let(:zendesk_ticket) { 'www.becomingateacher.zendesk.com/agent/tickets/example' }
+
+    it 'returns false if not valid' do
+      course_choice = create(:application_choice, status: :offer)
+      declined_offer_form = SupportInterface::ApplicationForms::ReinstateDeclinedOfferForm.new
+
+      expect(declined_offer_form.save(course_choice)).to eq(false)
+    end
+
+    it 'updates the provided ApplicationChoice with the "offer made" status if valid' do
+      Timecop.freeze do
+        course_choice = create(:application_choice, status: :declined)
```

Comment:
we should use `create(:application_choice, :with_declined_offer)` here otherwise line 48 will always return true